Teleinfo

guiguid opened the pull request:
## Description:

Add support for Teleinfo Serial Sensor.

Teleinfo is a French specific protocol used in electricity smart meters.
It provides real time information on power consumption, rates and current on
a user accessible serial port.

- Full asyncio version
- Only one deps (pyserial)

## Example entry for `configuration.yaml` (if applicable):
```yaml
 sensor:
   - platform: teleinfo
        name: "EDF teleinfo"
        serial_port: "/dev/ttyAMA0"
  - platform: template
    sensors:
      teleinfo_base:
        value_template: '{{ (states.sensor.edf_teleinfo.attributes["BASE"] | float / 1000) | round(0) }}'
        unit_of_measurement: 'kWh'
        icon_template: mdi:flash
  - platform: template
    sensors:
      teleinfo_iinst1:
        value_template: '{{ states.sensor.edf_teleinfo.attributes["IINST1"] | int }}'
        unit_of_measurement: 'A'
        icon_template: mdi:flash
```

## Checklist:
  - [X] The code change is tested and works locally.
  - [X] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [X] There is no commented out code in this PR.

If user exposed functionality or configuration variables are added/changed:
  - [X] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)
  -  [home-assistant/home-assistant.io#4382](https://github.com/home-assistant/home-assistant.io/pull/4382)

If the code communicates with devices, web services, or third-party tools:
  - [X] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [X] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [X] New or updated dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [X] New files were added to `.coveragerc`.

Linked PR (thanks to @nlamirault ): 
Add EDF Teleinfo sensor #14200
[WIP] Add: Teleinfo sensor #10603

guiguid committed:
Added Teleinfo Sensor

guiguid committed:
Added Teleinfo Sensor

guiguid committed:
Fix line length and back slash

guiguid committed:
Fix line length and back slash

guiguid committed:
Complete .covergaerc

guiguid committed:
Complete .covergaerc

guiguid committed:
Fix line length also in comments so breaking example ...

guiguid committed:
Fix line length also in comments so breaking example ...

homeassistant commented:
Hi @guiguid, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#20906).

Once you do that we will be able to review and accept this pull request.

Thanks!

wonderslug committed:
Set August doorbell availability state from online state (#20883)

The available state for the August Doorbell is currently set based on the state of the binary ding sensor.  This means that if there is no door bell rings in a while on startup the doorbell is shown as Unavailable (#20421) . This ties the availability of the doorbell to the  actual online state thats in the device information retrieved from august so that even if there has not been a doorbell ring activity on a while the device shows as online when it is.

guiguid committed:
Fix unsued var

guiguid committed:
Fix unsued var

VirtualL committed:
Added more language options (#20890)

This is a Small PR, Just Added 3 more language : 
* he: Hebrew
* ko: Korean
* lv: Latvian

guiguid committed:
Fix requirements_all

guiguid committed:
Fix requirements_all

balloob committed:
Person component: add storage and WS commands (#20852)

* Forbid duplicate IDs

* Allow loading persons from storage

* Convert to PersonManager

* Add storage support and WS commands to Person component

* Convert list command to differentiate types

* Allow loading person component without defining persons

* Fix cleanups after update/delete

* Address comments

* Start tracking when HA started

guiguid committed:
Fix last lint spaces

guiguid committed:
Fix last lint spaces

awarecan committed:
Fix coroutine never awaited warning in test (#20892)

guiguid committed:
Add Test file

guiguid committed:
Add Test file

guiguid committed:
fix flake8 unused import

guiguid committed:
fix flake8 unused import

dmulcahey committed:
Add some api tests for ZHA (#20909)

* start API tests

* blank lines

dagobert committed:
Missing Binary Sensor (#20921)

TiltIP Binary Sensor is missing in the discover list.

emontnemery committed:
Fix encoding for MQTT camera (#20932)

javicalle committed:
Fix RFLink restore state  (#20588)

* some minor tests refactor

* unused import

* async/await refactor

* Correct tests failures

* Restore state bug
added call to super().async_added_to_hass()

* Show brightness attribute if device supports it

* Fix light/test_rflink 

Dimmable devices defaults to 255 brightness

* delete super().device_state_attributes call

peternijssen committed:
add fan support for spider thermostats (#20897)

* add fan support for spider thermostats

* resolved feedback on pull request

guiguid committed:
Merge branch 'teleinfo' of https://github.com/guiguid/home-assistant into teleinfo

guiguid commented:
sorry, 
I'm not a git expert .... 
git rebase error that i don't know how to solve ....
I'll make another PR.
